### PR TITLE
clients: improve ci

### PR DESCRIPTION
**Pull request by octobocto**
This PR:
- Adds format lint and testing for all clients
- Moves building sidesail into a separate workflow
- Adds dependabot to drivechain_client and faucet_client

### CODE REVIEW

File: packages/drivechain_client/Justfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+lint:
+   find . -name "*.dart" -not -path "./lib/gen/*" | xargs dart format -l 120
```
Comment (torkelrogstad) [inline packages/drivechain_client/Justfile:2]: this seems more like formatting?

File: packages/faucet_client/Justfile [hunk 2]
```diff
@@ -1,2 +1,5 @@
 gen:
-    buf generate
\ No newline at end of file
+    buf generate
+
+lint:
+    find . -name "*.dart" -not -path "./lib/gen/*" | xargs dart format -l 120
```
Comment (torkelrogstad) [inline packages/faucet_client/Justfile:5]: this also seems like formatting

File: packages/sail_ui/Justfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+lint:
+   find . -name "*.dart" -not -path "./lib/gen/*" | xargs dart format -l 120
```
Comment (torkelrogstad) [inline packages/sail_ui/Justfile:2]: formatting!

File: packages/sidesail/Justfile [hunk 4]
```diff
@@ -0,0 +1,2 @@
+lint:
+   find . -name "*.dart" -not -path "./lib/gen/*" | xargs dart format -l 120
```
Comment (torkelrogstad) [inline packages/sidesail/Justfile:2]: foooormatting

Comment (torkelrogstad) [reply packages/sidesail/Justfile:2]: would it be better to have a dart formatting task in a root `Justfile`?

Comment (octobocto) [reply packages/sidesail/Justfile]: not yet. Maybe later if we want to get even fancier. most of the time you work with one specific package

**Pull request merged**